app/workers/scheduler/email_domain_block_refresh_scheduler.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Scheduler::EmailDomainBlockRefreshScheduler
+  include Sidekiq::Worker
+  include Redisable
+
+  sidekiq_options retry: 0
+
+  def perform
+    Resolv::DNS.open do |dns|
+      dns.timeouts = 5
+
+      EmailDomainBlock.find_each do |email_domain_block|
+        ips = begin
+          if is_ip?(email_domain_block.domain)
+            [email_domain_block.domain]
+          else
+            dns.getresources(email_domain_block.domain, Resolv::DNS::Resource::IN::A).to_a + dns.getresources(email_domain_block.domain, Resolv::DNS::Resource::IN::AAAA).to_a.map { |resource| resource.address.to_s }
+          end
+        end
+
+        email_domain_block.update(ips: ips, last_refresh_at: Time.now.utc)
+      end
+    end
+  end
+
+  def is_ip?(str)
```

Comment:
```suggestion
          if ip?(email_domain_block.domain)
            [email_domain_block.domain]
          else
            dns.getresources(email_domain_block.domain, Resolv::DNS::Resource::IN::A).to_a + dns.getresources(email_domain_block.domain, Resolv::DNS::Resource::IN::AAAA).to_a.map { |resource| resource.address.to_s }
          end
        end

        email_domain_block.update(ips: ips, last_refresh_at: Time.now.utc)
      end
    end
  end

  def ip?(str)
```

<img width="608" alt="image" src="https://user-images.githubusercontent.com/12539/155520627-f99113f1-7071-4e21-b9b9-3ddaf931d69d.png">